engine/apps/alerts/models/alert_group.py

```diff
@@ -896,10 +905,10 @@ def attach_by_user(self, user: User, root_alert_group: "AlertGroup", action_sour
                 action_source=action_source,
             )
 
-    def un_attach_by_user(self, user: User, action_source: Optional[str] = None) -> None:
+    def un_attach_by_user(self, user: User, action_source: typing.Optional[str] = None) -> None:
         AlertGroupLogRecord = apps.get_model("alerts", "AlertGroupLogRecord")
 
-        root_alert_group = self.root_alert_group
+        root_alert_group: AlertGroup = self.root_alert_group
```

Comment:
presumably we are **positive** that `self.root_alert_group is not None` at this point because we do things like `root_alert_group.log_records.create` a few lines down